pkg/dns/aws/dns.go

```diff
@@ -308,6 +309,14 @@ func (m *Manager) updateAlias(domain, zoneID, target, targetHostedZoneID, action
 		},
 	})
 	if err != nil {
+		if action == string(deleteAction) {
+			if aerr, ok := err.(awserr.Error); ok {
+				if strings.Contains(aerr.Message(), "not found") {
```

Comment:
Ouch, so there's no specific error type to check for this?

Reply:
https://docs.aws.amazon.com/sdk-for-go/api/service/route53/#Route53.ChangeResourceRecordSets

>* ErrCodeNoSuchHostedZone "NoSuchHostedZone"
>No hosted zone exists with the ID that you specified.

Is this the error type we're interested in?